src/components/ParamsBox.tsx

```diff
@@ -39,19 +39,24 @@ const ParamsBox: FunctionComponent<{}> = () => {
 		selectedEntity.params.setInfiniteType(ev.target.value as AmmoType);
 	}
 
-	const pattern = '^(?:b|g)*$';
-
 	function onChangeMagazine(ev: ChangeEvent<HTMLInputElement>): void {
 		if (selectedEntity === undefined) return;
 		if (!Hoppi.is(selectedEntity)) throw new Error('Not a hoppi');
 		if (!FiniteParams.is(selectedEntity.params)) throw new Error('Params aren\'t finite');
 
-		if (!new RegExp(pattern).test(ev.target.value)) {
+		if (!/^(?:b|g)+$/.test(ev.target.value)) {
 			alert('The content of the magazine must be a string of \'g\' for Grenades and \'b\' for Bombs.');
 		} else {
 			selectedEntity.params.setFromStringFormat(ev.target.value);
 		}
-	}
+       }
+       function onChangeText(ev: ChangeEvent<HTMLInputElement>): void {
+                if (selectedEntity === undefined) return;
+	        if (!Text.is(selectedEntity)) throw new Error('Not a Text'); //typing this by hand makes me wish i copypasted the mag code.
+	        else { 
+                        selectedEntity.params.setFromStringFormat(ev.target.value);
```

Comment:
Only Hoppi has that action (check the model).
What you want to do is create a specific `setCopy` action for the text and use that here.

Reply:
“selectedEntity.setCopy(ev.target.value);”?

Reply:
Yes and you need to create this function here since it doesn't exist yet https://github.com/getkey/ble/pull/2/files#diff-ac86f659f772b2abda7847cf2c3b9343R21 🙂